assets/src/stories-editor/helpers/index.js

```diff
@@ -1719,3 +1744,222 @@ export const processMedia = ( media ) => {
 		poster,
 	};
 };
+
+/**
+ * Returns a movable block's wrapper element.
+ *
+ * @param {Object} block Block object.
+ *
+ * @return {null|Element} The inner element.
+ */
+export const getBlockWrapperElement = ( block ) => {
+	if ( ! block ) {
+		return null;
+	}
+
+	const { name, clientId } = block;
+	const isMovableBlock = ALLOWED_MOVABLE_BLOCKS.includes( name );
+
+	if ( ! isMovableBlock ) {
+		return null;
+	}
+
+	return document.querySelector( `.amp-page-child-block[data-block="${ clientId }"]` );
+};
+
+/**
+ * Calculate target scaling factor so that it is at least 25% larger than the
+ * page.
+ *
+ * @param {number} width Target width.
+ * @param {number} height Target height.
+ *
+ * @return {number} Target scaling factor.
+ */
+const calculateTargetScalingFactor = ( width, height ) => {
+	const targetFitsWithinPage = width <= STORY_PAGE_INNER_WIDTH || height <= STORY_PAGE_INNER_HEIGHT;
+
+	if ( targetFitsWithinPage ) {
+		const scalingFactor = 1.25;
+
+		const widthFactor = STORY_PAGE_INNER_WIDTH > width ? STORY_PAGE_INNER_WIDTH / width : 1;
+		const heightFactor = STORY_PAGE_INNER_HEIGHT > height ? STORY_PAGE_INNER_HEIGHT / height : 1;
+
+		return Math.max( widthFactor, heightFactor ) * scalingFactor;
+	}
```

Comment:
But what if the element is already larger than the page, but only by a tiny amount, say 1%? Then this function would suggest not to scale the element, even thought it's smaller than the required size, no?

Maybe some unit tests would be helpful for this simple utility function.

Reply:
FWIW, this method is basically a straight copy from the AMP framework to make sure the logic is the same: https://github.com/ampproject/amphtml/blob/13b3b6d92ee0565c54ec34732e88f01847aa8a91/extensions/amp-story/1.0/animation-presets-utils.js#L91-L111